src/libs/addEmojiToComposer/index.android.js

```diff
@@ -0,0 +1,29 @@
+import baseAddEmojiToComposer from './baseAddEmojiToComposer';
+
+/**
+ * Takes a text and adds an emoji at the place of selection.
+ * It will then update the text of the given TextInput using its `setTextAndSelection` method.
+ * `setTextAndSelection` method is usually available on TextInput refs from the composer component.
+ *
+ * @param {Object} params
+ * @param {String} params.text The text where the emoji should be added
+ * @param {String} params.emoji The emoji to add
+ * @param {Object} params.textInput
+ * @param {{start: Number, end: Number}} params.selection
+ *
+ * @return {{ newSelection: {start: Number, end: Number}, newText: String }} results
+ */
+function addEmojiToComposerTextInput(params) {
+    const {selection, textInput} = params;
+    const hasRangeSelected = selection.start !== selection.end;
+    if (hasRangeSelected) {
+        // Android: when we have a range selected setSelection
```

Comment:
This seems like a React Native bug? If so, can we please create an upstream issue and start investing an upstream fix?

Reply:
yeah it is, but it might go away with fabric, so I wanted to check for that first

Reply:
Can you create an issue to follow-up then link it here? It will be automatically closed by if you link it here I'll reopen it and assign it to you. It's always good to follow-through with these cleanup-style items or we'll end up with a lot of technical debt

Reply:
https://github.com/Expensify/App/issues/12587